lib/krane/deprecated_deploy_task.rb

```diff
@@ -453,9 +453,11 @@ def replace_or_create_resource(resource, force: false)
         ["replace", "-f", resource.file_path]
       end
 
-      _, _, status = kubectl.run(*args, log_failure: false)
+      _, _, status = kubectl.run(*args, log_failure: false, output_is_sensitive: resource.sensitive_template_content?,
+        raise_if_not_found: true)
       return [nil, status] if status.success?
 
+    rescue Krane::Kubectl::ResourceNotFoundError
       # it doesn't exist so we can't replace it, we try to create it
       create_resource(resource)
```

Comment:
I think you need to explicitly `return` here--that's probably what's causing your CI failure